### ISSUE TITLE
A few bugs in firefox...

### ISSUE THREAD

**Issue by theoryshaw** [JohnMcLear/draw#212]
* pan, with middle button, doesn't work-- just stays on a drawing tool.
* carriage return in a text box, is not remember--snaps back to single line.
* zoom to extents doesn't seem to work.

**Comment by theelectricwiz** [comment]
Ah. Thanks. Hmm. The middle click pan one is a strange one. Will have to bit of investigation on that one. I have my suspicions (not good for getting working), but will have a look. As a workaround, there is the CTRL+drag for firefox.
The carriage return should be easy to fix.
Re the zoom to extent - what version of FF are you using? Does the zoom to extent button work (one with arrows) for you?

**Comment by theoryshaw** [comment]
VER. 42.0

yes, the zoom to extents button works.... just not double click on the middle button.

**Comment by theelectricwiz** [comment]
*clunk* Right. The firefox panning issue was one I had found already, hence I added the key modifier method (#202). Have submitted a bug report to Mozilla about that one https://bugzilla.mozilla.org/show_bug.cgi?id=1231221.
More bad news for the double middle click. Non-existent in firefox. Also reported a bug about that one https://bugzilla.mozilla.org/show_bug.cgi?id=1231223.
Apologies, the button and the CTRL+left drag are the best I can do for those atm.
